src/sentry/shared_integrations/response/base.py

```diff
@@ -10,28 +14,35 @@
 class BaseApiResponse:
     text = ""
 
-    def __init__(self, headers=None, status_code=None):
+    def __init__(
+        self,
+        headers: Mapping[str, str] | None = None,
+        status_code: int | None = None,
+    ) -> None:
         self.headers = headers
         self.status_code = status_code
 
-    def __repr__(self):
-        return "<{}: code={}, content_type={}>".format(
-            type(self).__name__,
-            self.status_code,
-            self.headers.get("Content-Type", "") if self.headers else "",
-        )
+    def __repr__(self) -> str:
+        content_type = (self.headers or {}).get("Content-Type", "")
+        return f"<{type(self).__name__}: code={self.status_code}, content_type={content_type}>"
+
+    @property
+    def json(self) -> Any:
+        raise NotImplementedError
 
-    @cached_property
-    def rel(self):
-        if not self.headers:
-            return {}
-        link_header = self.headers.get("Link")
-        if not link_header:
-            return {}
-        return {item["rel"]: item["url"] for item in requests.utils.parse_header_links(link_header)}
+    @cached_property  # type: ignore
```

Comment:
i feel like we ought to be able to load type stubs for Django to make this not necessary

Reply:
Yup, that's what I want to work on next :)